aip/client-libraries/4223.md

```diff
@@ -0,0 +1,39 @@
+---
+id: 4223
+state: draft # TODO(noahdietz): Update pre-merge
+created: 2022-05-11 # TODO(noahdietz): Update pre-merge
+---
+
+# Client-side paylod validation
+
+Services document the behavior of specific request fields using the
+`google.api.field_behavior` annotation defined in [AIP-203][]. This annotation
+conveys how a service interprets the field, including how a field may be
+validated (or not) by the service.
+
+## Guidance
+
+Client libraries **must not** implement client-side payload validation based
+on the `google.api.field_behavior` annotation, except to prevent
+[local failures](4223.md#local-failures).
+
+The `google.api.field_behavior` annotation is a primarily a machine-readable
```

Comment:
Should we have this and the following paragraphs under a section called "Motivation" or "Reasoning"? They explain why we're giving this guidance, but are not part of the guidance itself. (I'm fine with following AIP precedent on this.)

Reply:
Yeah this is tough because I wanted to document the _why_, but AIPs  don't have a pattern for sections like "Reasoning" as they are more "do this" rather than "here is a design doc". I put it under guidance and following the primary piece of guidance because I wanted to expand on it. I could see a subsection named `### Server-side validation` and explain why clients "must" depend on it (which implies why clients must not do it themselves). WDYT?

Reply:
Hmm, maybe we should start a precedent of adding a "motivation" section.

Less drastically, we could start this line with "The reason for this is that the google.api.field_behavior annotation ..."

Reply:
I went with the "less drastic" approach 😄